cmake/USDPluginTools.cmake

```diff
@@ -830,3 +793,69 @@ function(_usd_target_properties
             ${_LINK_LIBRARIES}
     )
 endfunction() # _usd_target_properties
+
+# Set-up runtime environment variables for the test.
+# When USE_PYTHONPATH is one, will include project python libraries
+function(_usd_set_tests_properties
+    TARGET_NAME
+    USE_PYTHONPATH
+)
+    # The paths refer to the build tree (which mirrors the final installation).
+    # The first path for an env var needs the 'VARNAME=' prepended.
+    # On windows, calls to "$ENV{}" must be sanitized to replace back slashes with forward slashes
+    # Env vars after the first must be prepended with a double escaped semicolon to be recognized. 
+
+    # TEST_PXR_PLUGINPATH_NAME
+    set(TEST_PXR_PLUGINPATH_NAME "$ENV{PXR_PLUGINPATH_NAME}")
+    if (MSVC)
+        string(REGEX REPLACE "\\\\" "/" TEST_PXR_PLUGINPATH_NAME "${TEST_PXR_PLUGINPATH_NAME}")
+    endif()
+    string(PREPEND TEST_PXR_PLUGINPATH_NAME "PXR_PLUGINPATH_NAME=")
+    list(APPEND TEST_PXR_PLUGINPATH_NAME
+        "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/${USD_PLUG_INFO_ROOT_DIR}"
+        "${PROJECT_BINARY_DIR}/${USD_PLUGIN_DIR}/${USD_PLUG_INFO_ROOT_DIR}"
+    )        
+
+    # TEST_PYTHON_PATH
+    if (USE_PYTHONPATH)
+        set(TEST_PYTHON_PATH "$ENV{PYTHONPATH}")
+        if (MSVC)
+            string(REGEX REPLACE "\\\\" "/" TEST_PYTHON_PATH "${TEST_PYTHON_PATH}")
+        endif()
+        string(PREPEND TEST_PYTHON_PATH "PYTHONPATH=")
+        string(PREPEND TEST_PYTHON_PATH "\\;")
+        list(APPEND TEST_PYTHON_PATH 
+            "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python"
+            "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python"
+        )
+
+        list(APPEND TEST_PXR_PLUGINPATH_NAME "${TEST_PYTHON_PATH}")
```

Comment:
Hey @colevfx, I wonder if there's a more generic variable name for `TEST_PXR_PLUGINPATH_NAME` - given that we're appending TEST_PYTHON_PATH to it as well.  Maybe ENV_VARS or something to that effect?  Open to suggestions!

Reply:
You know it's funny I went back and forth on this a lot! In this situation, adding the `ENV_VARS` thing as a generic cmake variable at the top would be redundant, since `PXR_PLUGINPATH_NAME` is an absolute requirement on all platforms. 

But you are totally right that we are mangling the `TEST_PXR_PLUGINPATH_NAME` variable, and while we don't do it here, it would be impossible to use the variable as intended if we parent scoped it or something. I'll see if I can make this a bit better 😄  I'm now thinking I should use a variable named `TEST_ENV_VARS`, to be a little more clear about what's going on!